app/models/concerns/profile_policy_association.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Methods that are related to a profile's policy
+module ProfilePolicyAssociation
+  extend ActiveSupport::Concern
+
+  included do
+    after_destroy :destroy_policy_test_results
+
+    def policy
+      return self unless external
+
+      Profile.includes(:benchmark)
+             .find_by(account: account_id, external: false, ref_id: ref_id,
+                      benchmarks: { ref_id: benchmark.ref_id })
+    end
+
+    def policy_profiles
+      Profile.includes(:benchmark)
+             .where(account: account_id, ref_id: ref_id,
+                    benchmarks: { ref_id: benchmark.ref_id })
+    end
+
+    def business_objective
+      BusinessObjective.find(business_objective_id) if business_objective_id
+    end
+
+    def business_objective_id
+      (policy || self).read_attribute(:business_objective_id)
+    end
+
+    def compliance_threshold
+      (policy || self).read_attribute(:compliance_threshold)
+    end
+
+    def destroy_policy_test_results
+      policy_profiles.destroy_all
```

Comment:
This is potentially OOM, if it has to destroy a couple of 1000s test results, can we do this asynchronously? Something like `app/jobs/delete_host.rb` would work